### PR TITLE
docs(prototypes): DLT-1740 left bar prototype

### PR DESCRIPTION
**Pull request by juliodialpad**
# Dialpad's Left bar prototype

## Obligatory GIF (super important!)

Obligatory GIF

## :hammer_and_wrench: Type Of Change

These types will not increment the version number, but will still deploy to documentation site on release:

- [x] Documentation

## :book: Jira Ticket

[DLT-1740](https://dialpad.atlassian.net/browse/DLT-1740)

## :book: Description

Added Dialpad's leftbar prototype and layout for the future when we want to start adding the rest of the app prototypes.

## :bulb: Context

We need a prototype of how the app should be build with dialtone components, following the correct semantics and fully accessible through screen readers and keyboard navigation.

## :pencil: Checklist

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [x] I have reviewed my changes.

For all Vue changes:

- [ ] I have made my changes in Vue 2 and Vue 3. Note: you may sync your changes from Vue 2 to Vue 3 (or vice versa) using the `./scripts/dialtone-vue-sync.sh` script. Read docs here: [Dialtone Vue Sync Script](../packages/dialtone-vue3/.github/CONTRIBUTING.md#dialtone-vue-sync-script)
- [x] I have validated components with a screen reader.
- [x] I have validated components keyboard navigation.

## :camera: Screenshots / GIFs

Screenshot 2024-06-14 at 9 45 37 a m

## :link: Sources

https://www.figma.com/design/oTY56LgQha8ljb4c8w4GtO/Dialpad-App-Components?node-id=9240-7131&m=dev

[DLT-1740]: https://dialpad.atlassian.net/browse/DLT-1740?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/dialtone-vue3/.storybook/preview.jsx
Comment (juliodialpad) [inline packages/dialtone-vue3/.storybook/preview.jsx:24]: faker global seed, to make sure results are reproducible on percy and don't change on every reload too.

File: packages/dialtone-vue3/components/list_item/list_item.vue
Comment (juliodialpad) [inline packages/dialtone-vue3/components/list_item/list_item.vue]: This was emitting a warning

File: packages/dialtone-vue3/prototypes/dialpad/leftbar/components/leftbar_section.vue
Comment (juliodialpad) [inline packages/dialtone-vue3/prototypes/dialpad/leftbar/components/leftbar_section.vue]: Had to put this d-w128 to achieve truncate, we'll need to make this with JS otherwise if we need it to be dynamically updated as the slot can be anything.

File: packages/dialtone-vue3/prototypes/dialpad/dialpad.stories.js
Comment (ninamarina) [inline packages/dialtone-vue3/prototypes/dialpad/dialpad.stories.js:6]: Why call it Dialpad instead of Leftbar?

File: packages/dialtone-vue3/prototypes/dialpad/leftbar/components/leftbar_section.vue
Comment (iropolo) [reply packages/dialtone-vue3/prototypes/dialpad/leftbar/components/leftbar_section.vue]: I would put this on a `<!-- -->` in the template, kind of a to do or just to let it know this what you are saying and not missing it in the PR.

File: packages/dialtone-vue3/prototypes/dialpad/leftbar/components/leftbar_section.vue
Comment (iropolo) [inline packages/dialtone-vue3/prototypes/dialpad/leftbar/components/leftbar_section.vue]: `defineProps` not necessary to import.

File: packages/dialtone-vue3/prototypes/dialpad/leftbar/components/options_dropdown.vue
Comment (iropolo) [inline packages/dialtone-vue3/prototypes/dialpad/leftbar/components/options_dropdown.vue]: `defineEmits` not necessary to import.

File: packages/dialtone-vue3/prototypes/dialpad/leftbar/dialpad_leftbar.vue
Comment (iropolo) [inline packages/dialtone-vue3/prototypes/dialpad/leftbar/dialpad_leftbar.vue]: Since all this is static data, I guess these can go without `ref`

File: packages/dialtone-vue3/.storybook/preview.jsx
Comment (iropolo) [reply packages/dialtone-vue3/.storybook/preview.jsx:24]: I would add this comment on the top of the `faker.seed`

File: packages/dialtone-vue3/prototypes/dialpad/dialpad.stories.js
Comment (iropolo) [inline packages/dialtone-vue3/prototypes/dialpad/dialpad.stories.js:19]: Maybe we can rename this `Default` to `Leftbar` then this will be the structure:

image

Just as a suggestion, related to Nina's comment

File: packages/dialtone-vue3/prototypes/dialpad/leftbar/components/options_dropdown.vue
Comment (iropolo) [inline packages/dialtone-vue3/prototypes/dialpad/leftbar/components/options_dropdown.vue:70]: Regarding events through the component,
We are emitting them but not showing it in the `Actions` tab of storybook, also we are not using it at all, Im guessing we can remove them or keep it track in the `Actions` tab.

Comment (iropolo) [reply packages/dialtone-vue3/prototypes/dialpad/leftbar/components/options_dropdown.vue:70]: Since it is a prototype I guess we can keep it track in the `Actions` tab to let us know which events are being fired through the usage.

Not so sure, open to discussion, it's fine if we leave it as it is now.

**Comment by iropolo** [comment]
Seems like DND or OOO is missing a transparency

Current:
image


Product:
image

**Comment by iropolo** [comment]
Not sure if it is a behaviour that we should replicate because this is just a design perspective, ~~but maybe it is... not sure about the scope of the prototypes 🤔 .~~

~~If you have unread msg in some collapsed section, that user conversation/channel or group chat will remain visible.~~

update edit: not in prototype scope


Screenshot 2024-06-18 at 13 55 55

**Review by francisrupert — COMMENTED** [review]
Pretty darn cool!

Each section header definitely too big, and expand/collapse chevron icon should align with other icons at its same vertical level.

### Prototype
https://github.com/dialpad/dialtone/assets/1165933/fec83e8f-adcb-4772-8b8a-6e070961491b

### Beta
https://github.com/dialpad/dialtone/assets/1165933/d72d4f23-d78b-4311-b368-eaccfd4d7609

File: packages/dialtone-vue3/prototypes/dialpad/dialpad.stories.js
Comment (braddialpad) [reply packages/dialtone-vue3/prototypes/dialpad/dialpad.stories.js:6]: I think because by the time it's done it will be a prototype of the entire core dialpad app.

File: packages/dialtone-vue3/components/list_item/list_item.vue
Comment (braddialpad) [reply packages/dialtone-vue3/components/list_item/list_item.vue]: What was the warning? There might be a problem with doing this because a list item can be non-interactive in which case we don't want to emit a click event.

File: packages/dialtone-vue3/prototypes/dialpad/leftbar/components/leftbar_section.vue
Comment (braddialpad) [reply packages/dialtone-vue3/prototypes/dialpad/leftbar/components/leftbar_section.vue]: does max-width work better? or it needs to be fixed?

File: packages/dialtone-vue3/prototypes/dialpad/leftbar/components/options_dropdown.vue
Comment (braddialpad) [reply packages/dialtone-vue3/prototypes/dialpad/leftbar/components/options_dropdown.vue:70]: Hmm probably okay to not emit them in a prototype in an internal component like this.

File: packages/dialtone-vue3/prototypes/dialpad/leftbar/dialpad_leftbar.vue
Comment (braddialpad) [inline packages/dialtone-vue3/prototypes/dialpad/leftbar/dialpad_leftbar.vue:1]: Nice, this is so clean looking compared to the current implementation in DP.

File: packages/dialtone-vue3/components/list_item/list_item.vue
Comment (juliodialpad) [reply packages/dialtone-vue3/components/list_item/list_item.vue]: The error was that it was emitting a click event but click wasn't declared on 'emits'

File: packages/dialtone-vue3/prototypes/dialpad/dialpad.stories.js
Comment (juliodialpad) [reply packages/dialtone-vue3/prototypes/dialpad/dialpad.stories.js:6]: Yeah, exactly, it's not just the leftbar, it's the entire dialpad prototype

File: packages/dialtone-vue3/prototypes/dialpad/dialpad.stories.js
Comment (juliodialpad) [reply packages/dialtone-vue3/prototypes/dialpad/dialpad.stories.js:19]: The prototype will be the entire dialpad app, not only the leftbar, right now it only includes the leftbar but will include everything else in the near future.

File: packages/dialtone-vue3/prototypes/dialpad/leftbar/components/leftbar_section.vue
Comment (juliodialpad) [reply packages/dialtone-vue3/prototypes/dialpad/leftbar/components/leftbar_section.vue]: Yes! max-width worked better indeed. 

I would like a design review here, is this ok to do? or should we set a fixed percent of it for the channel name and the rest for the button?

Something like 60/40 instead of a fixed 12.8 rem? I think the problem is that on product the leftbar is resizable.

Screenshot 2024-06-20 at 12 07 21 p m
Screenshot 2024-06-20 at 12 07 42 p m

@francisrupert @fede-dp

File: packages/dialtone-vue3/prototypes/dialpad/leftbar/components/options_dropdown.vue
Comment (juliodialpad) [reply packages/dialtone-vue3/prototypes/dialpad/leftbar/components/options_dropdown.vue:70]: Yes, that's what I was thinking, we don't really need to emit the events in this component. I'll left the component as is right now.

File: packages/dialtone-vue3/components/list_item/list_item.vue
Comment (juliodialpad) [reply packages/dialtone-vue3/components/list_item/list_item.vue]: https://github.com/dialpad/dialtone/assets/87546543/a1b4d88c-a8dd-490b-b55e-478e748e3323

Only firing the event while pressing `enter` or `space` on a list-item

**Review by iropolo — APPROVED** [review]
Looks good to me!,
Lets wait for Design approval. 

Vue2 version would be in a separeted PR?, thanks

**Comment by iropolo** [comment]
Maybe there is a slightly improvement for the structure, since it will hold all the Dialpad structure and not just the Leftbar.
Could we create a `section` folder, move in there `Leftbar` folder.
There is also a component that will be probably shared trough others sections like `options_dropdown.vue`, create a components folder at the same level of `section`

Will be sth like:

```
prototypes/
├─ dialpad/
│  ├─ sections/
│  │  ├─ leftbar/
│  │  ├─ another_section/
│  ├─ components/
│  │  ├─ options_dropdown.vue/

```

I have some concerns about the static data generated by `faked`, for now seems to work softly but is there a reason that we want to have some dynamic data in the prototype? 
I was thinking maybe having a separeted folder to hold all this static data.

This will probably mutate in the near future.
Not necessary to do anything of this now, we can wait to see how it grows.

**Comment by fede-dp** [comment]
looks good to me, at least visually! let's wait for Francis to check the semantics

File: packages/dialtone-vue3/prototypes/dialpad/leftbar/components/options_dropdown.vue
Comment (iropolo) [inline packages/dialtone-vue3/prototypes/dialpad/leftbar/components/options_dropdown.vue]: This probably not need to be a `ref`, seems like it is static data

**Comment by juliodialpad** [comment]
> Maybe there is a slightly improvement for the structure, since it will hold all the Dialpad structure and not just the Leftbar. Could we create a `section` folder, move in there `Leftbar` folder. There is also a component that will be probably shared trough others sections like `options_dropdown.vue`, create a components folder at the same level of `section`
> 
> Will be sth like:
> 
> ```
> prototypes/
> ├─ dialpad/
> │  ├─ sections/
> │  │  ├─ leftbar/
> │  │  ├─ another_section/
> │  ├─ components/
> │  │  ├─ options_dropdown.vue/
> ```
Sounds like a good improvement, but I'd prefer to keep it as it is right now and mutate when needed in the future, maybe the requirements change, so let's not double the work now and later.

> I have some concerns about the static data generated by `faked`, for now seems to work softly but is there a reason that we want to have some dynamic data in the prototype? I was thinking maybe having a separated folder to hold all this static data.

We want dynamic data on the prototype to check how it behaves on different cases, with long and short texts, we might also want some status with emojis or something like that in the future, all of that can be achieved with `faker`.

You mean to have a separate folder with some JSON files with this data? I feel like using faker in this way is more natural and automated, which is the advantage of doing the way you're thinking?

**Comment by iropolo** [comment]
> You mean to have a separate folder with some JSON files with this data? I feel like using faker in this way is more natural and automated, which is the advantage of doing the way you're thinking?

All static data should be separated from component logic, since it is not required to be there.
Based on that, there should be room for improvement.

For now it can still working as this way, we just have to pay attention to scalability and to have a readable structure.

**Pull request merged**
